src/main/java/terminallayer/TerminalMain.java

```diff
@@ -0,0 +1,198 @@
+import datamodels.Card;
+import datamodels.User;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import terminallayer.exceptions.InvalidTypeException;
+
+import java.math.BigDecimal;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Scanner;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static terminallayer.util.TerminalUtil.*;
+
+
+public class TerminalMain {
+
+    private static final Logger LOGGER = LogManager.getLogger("TESTLOGGER");
+    private static final Scanner scan = new Scanner(System.in);
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException, InvalidTypeException, SQLException, DAOException {
+        while (true) {
+            LOGGER.info("date");
+            String date = scan.nextLine();
+            if (date.matches("[0-9][0-9]-[0-9][0-9]")) {
+                LOGGER.info("passed");
+            }
+            LOGGER.info("ATM Terminal \n");
+            LOGGER.info("1) User");
+            LOGGER.info("2) Admin");
+            LOGGER.info("3) Create Account");
+            LOGGER.info("4) Exit");
+            switch (scan.nextInt()) {
+                case 1:
+                    Account a = authUser();
+                    userUI(a);
+                    break;
+                case 2:
+                    authUser();
+                    adminUI();
+                    break;
+                case 3:
+                    createAccount();
+                    break;
+                case 4:
+                    LOGGER.info("Have A Good Rest of Your Day!");
+                    scan.close();
+                    break;
+                default:
+                    throw new InvalidTypeException("Invalid Option");
+            }
+        }
+    }
+
+    public static void userUI(Account a) throws InvalidTypeException, SQLException, DAOException {
+        AccountsDAO accountsDAO = new AccountsDAO();
+        CardsDAO cardsDAO = new CardsDAO();
+        LOGGER.info("1) Check Account ");
+        LOGGER.info("2) Deposit Money");
+        LOGGER.info("3) Transfer Money");
+        LOGGER.info("4) Withdraw money");
+        LOGGER.info("5) Report Card Stolen");
+        Card c = cardsDAO.getById(a.getId());
+        switch (scan.nextInt()) {
+            case 1:
+                LOGGER.info("account ");
+                LOGGER.info(a);
+                break;
+            case 2:
+                LOGGER.info("How Much money would you like to deposit: ");
+                BigDecimal deposit = scan.nextBigDecimal();
+                accountsDAO.makeDeposit(a, deposit);
+                break;
+            case 3:
+                makeTransfer(a);
+                break;
+            case 4:
+                LOGGER.info("Amount to Withdraw");
+                BigDecimal withdraw = scan.nextBigDecimal();
+                accountsDAO.makeWithdrawal(a, withdraw);
```

Comment:
For both makeTransfer and makeWithdrawal you should catch the DAOException (which is thrown when an account has insufficient funds) and inform the user of the issue.